core/shared/src/main/scala/spire/math/package.scala

```diff
@@ -404,10 +404,11 @@ package object math {
     (implicit f: Field[A], n: NRoot[A], s: Signed[A]): A = {
     import spire.implicits._
     def abs(n: A): A = if (n < f.zero) -n else n
-    if (x == f.zero) y
-    else if (y == f.zero) x
-    else if (x > y) abs(x) * (1 + (y/x)**2).sqrt
-    else abs(y) * (1 + (x/y)**2).sqrt
+    val (ax, ay) = (abs(x), abs(y))
```

Comment:
Is the `Tuple2` allocation/pattern matching optimized away?

I'd prefer to see `val ax = abs(x); val ay = abs(y)`, especially in specialized methods.

Reply:
good point, I wouldn't be surprised if it's not, just fixed